generate_brick.py

```diff
@@ -86,90 +103,70 @@ def add_tags(klass, definition):
-        G.add( (BRICK[rootclass], A, OWL.Class) )
-        G.add( (BRICK[rootclass], RDFS.subClassOf, BRICK.Class) )
-        for k, v in properties.items():
-            if isinstance(v, list) and k == "tags":
-                add_tags(rootclass, v)
-            elif isinstance(v, list) and k == "substances":
-                add_class_restriction(rootclass, v)
-            elif not apply_prop(rootclass, k, v):
-                if isinstance(v, dict) and k == "subclasses":
-                    define_subclasses(v, BRICK[rootclass])
+def define_classes(definitions, parent, pun_classes=False):
+    """
+    Generates triples for the hierarchy given by 'definitions', rooted
+    at the class given by 'parent'
+    - class hierarchy ('subclasses')
+    - tag mappings
+    - substance + quantity modeling
+
+    If pun_classes is True, then create punned instances of the classes
+    """
+    for classname, defn in definitions.items():
+        classname = BRICK[classname]
+        # class is a owl:Class
+        G.add((classname, A, OWL.Class))
+        # subclass of parent
+        G.add((classname, RDFS.subClassOf, parent))
+        # add label
+        class_label = classname.split('#')[-1].replace("_", " ")
+        G.add((classname, RDFS.label, Literal(class_label)))
+        if pun_classes:
+            G.add((classname, A, classname))
+
+        # define mapping to tags if it exists
+        # "tags" property is a list of URIs naming Tags
+        taglist = defn.get('tags', [])
+        assert isinstance(taglist, list)
+        if len(taglist) == 0:
+            logging.warning(f"Property 'tags' not defined for {classname}")
+        add_tags(classname, taglist)
+
+        # define mapping to substances + quantities if it exists
+        # "substances" property is a list of (predicate, object) pairs
+        substancedef = defn.get('substances', [])
+        assert isinstance(substancedef, list)
+        add_restriction(classname, substancedef)
+
+        # define class structure
+        # this is a nested dictionary
+        subclassdef = defn.get('subclasses', {})
+        assert isinstance(subclassdef, dict)
+        define_classes(subclassdef, classname, pun_classes=pun_classes)
+
+        # handle 'parents' subclasses (links outside of tree-based hierarchy)
+        parents = defn.get('parents', [])
+        assert isinstance(parents, list)
+        for _parent in parents:
+            G.add((classname, RDFS.subClassOf, _parent))
+
+        # all other key-value pairs in the definition are
+        # property-object pairs
+        expected_properties = ['parents', 'tags', 'substances', 'subclasses']
+        other_properties = [prop for prop in defn.keys()
+                            if prop not in expected_properties]
+        for propname in other_properties:
+            propval = defn[propname]
+            G.add((classname, propname, propval))
 
 
 def apply_prop(prop, pred, obj):
```

Comment:
Shouldn't this be removed based on the changes?

Reply:
This adds the "miscellaneous" properties in: OWL.equivalentClass, SKOS.definition, etc. The first part of the method handles the "special cases" where the keys are not URIs

Reply:
But it is not used anywhere in the code. Is it used somewhere else?

Reply:
Which part specifically isn't used in the code? All definitions should be going through this `define_classes` method

Reply:
btw, I am talking about `apply_prop`.

Reply:
Oh, your comment is on the `define_classes` method. You are correct I forgot to remove `apply_prop`. Addressed in the latest commit

Reply:
I added a comment to the exact line and Github focused on the changes I think. Thanks for the fix.